### PR TITLE
adding deployment to S3 bucket for staging

### PR DESCRIPTION
**Pull request by sebbsan**

### CODE REVIEW

**Comment by ajantsch** [comment]
Staging Deployment Successful, check it out on this URL: gesundzusammen-staging.finleap.tech.s3.eu-central-1.amazonaws.com/index.html

**Comment by sebbsan** [comment]
staging thing setup, but there is a problem: it will right now trigger a staging delpoyment for every push to a branch that’s not master or source. Which means parallel branches will overwrite each others. We have these options:
create new S3 bucket for each new deployment -> but this will create many many buckets without cleaning them up.. but maybe there can be a github action running after successful merge to master to delete the bucket…
deploy into bucket subfolder for each new branch -> tried that, right now the App is redirecting back to the root folder / host. So needs a change in the App
adjust our git workflow - we could merge all PRs into a new default staging branch which will trigger the action. After reviewing on staging, someone needs to merge from staging into source

**Comment by ajantsch** [comment]
Staging Deployment Successful, check it out on this URL: gesundzusammen-staging.finleap.tech.s3.eu-central-1.amazonaws.com/index.html

File: infrastructure/terraform/terraform.tfstate.backup [hunk 1]
```diff
@@ -0,0 +1,200 @@
+{
```
Comment (sebbsan) [inline infrastructure/terraform/terraform.tfstate.backup:1]: woops I need to delete this and add to gitignore

File: infrastructure/terraform/variables.tf [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+variable "aws-secret-key" {
+  type = string
+}
+
+variable "bucket_name" {
+  description = "name of the S3 bucket where the static website will be hosted, e.g. \"demo.com\""
+  type        = string
+  default     = "gesundzusammen-staging.finleap.tech"
```
Comment (manzke) [inline infrastructure/terraform/variables.tf:24]: should we paste this via a secret variable?

File: infrastructure/terraform/variables.tf [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+variable "domain_name" {
+  description = "domain for the cloudfront distribution e.g. \"demo.com\""
+  type        = string
+  default     = "gesundzusammen-staging.finleap.tech"
```
Comment (manzke) [inline infrastructure/terraform/variables.tf:30]: same here - should we paste this via a secret variable?

File: infrastructure/terraform/variables.tf (hunk 2 above)
Comment (sebbsan) [reply infrastructure/terraform/variables.tf]: ok I removed the reference to staging URL and bucket

File: infrastructure/terraform/variables.tf (hunk 3 above)
Comment (sebbsan) [reply infrastructure/terraform/variables.tf]: ok I removed the reference to staging URL and bucket

**Pull request merged**